### PR TITLE
M5014 molgenis-data modularity improvements

### PR DESCRIPTION
**Pull request by dennishendriksen**
#### Checklist
- [ ] Functionality works & meets specifications
- [ ] Code reviewed
- [ ] Code unit/integration/system tested
- [ ] User documentation updated
- [ ] (If you have changed REST API interface) view-swagger.ftl updated
- [ ] Test plan template updated
- [ ] Clean commits

### CODE REVIEW

**Comment by fdlk** [comment]
Why do we want/need to move the GsonConfig up in the tree?
I'd expect the REST API to fail without it.

**Comment by dennishendriksen** [comment]
test this please

File: molgenis-data-elasticsearch/pom.xml [hunk 1]
```diff
@@ -26,16 +18,40 @@
...
-            <artifactId>molgenis-data</artifactId>
+            <artifactId>molgenis-data-index</artifactId>
             <version>${project.version}</version>
-            <type>test-jar</type>
-            <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.molgenis</groupId>
             <artifactId>molgenis-jobs</artifactId>
             <version>${project.version}</version>
```
Comment (dennishendriksen) [inline molgenis-data-elasticsearch/pom.xml]: Yes, but currently both IndexConfig (should be moved to molgenis-data-index) and EmbeddedElasticSearchServiceFactory (can be removed once we use standalone Elasticsearch) depend on it. Lets remove the dependency once we wrap up M4667).

File: molgenis-data-rest-client/pom.xml [hunk 2]
```diff
@@ -22,74 +25,37 @@
...
-            </exclusions>
+            <groupId>org.slf4j</groupId>
+            <artifactId>jcl-over-slf4j</artifactId>
         </dependency>
         <dependency>
-            <groupId>org.springframework</groupId>
-            <artifactId>spring-web</artifactId>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient</artifactId>
+            <version>4.3.5</version>
```
Comment (fdlk) [inline molgenis-data-rest-client/pom.xml]: Should be 4.5.1?

File: pom.xml [hunk 3]
```diff
@@ -581,121 +625,186 @@
...
+                <version>2.3.23</version>
+            </dependency>
             <!-- see http://docs.spring.io/spring-security/site/docs/3.2.x/reference/htmlsingle/#maven-bom -->
             <dependency>
                 <groupId>org.springframework</groupId>
                 <artifactId>spring-framework-bom</artifactId>
                 <version>${spring.version}</version>
                 <type>pom</type>
                 <scope>import</scope>
+                <!-- Enable exclusions when dependency management import supports exclusions: https://issues.apache.org/jira/browse/MNG-5600 -->
```
Comment (fdlk) [inline pom.xml]: File Github issue for this and remove the detailed commented out exclusions.

File: molgenis-data-rest-client/pom.xml (hunk 2 above)
Comment (dennishendriksen) [inline molgenis-data-rest-client/pom.xml]: Done

File: pom.xml (hunk 3 above)
Comment (dennishendriksen) [inline pom.xml]: I could file a GitHub issue, but we can't do anything about it until MNG-5600 is fixed. Personally I think this in-your-face comments will keep some of us checking regularly to see if this issue has been fixed.

Comment (fdlk) [inline pom.xml]: I've filed #6214 as a reminder.

**Pull request merged**